### PR TITLE
Provider: Add YNAB provider

### PR DESCRIPTION
**Pull request by cwilby**
API Documentation: https://api.ynab.com/
Endpoint Overview: https://api.ynab.com/v1#/

PR for website documentation: https://github.com/SocialiteProviders/website/pull/59

### CODE REVIEW

File: src/YNAB/Provider.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        return (new User())->setRaw($user)->map([
+            'id'       => $user['id'],
+            'nickname' => null,
+            'name'     => null,
+            'email'    => null,
+            'avatar'   => null
+        ]);
+    }
+}
```
Comment (atymic) [inline src/YNAB/Provider.php:78]: missing newline

File: src/YNAB/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+Please see the [Base Installation Guide](https://socialiteproviders.com/usage/), then follow the provider specific instructions below.
+
+### Add configuration to `config/services.php`
+
+```php
+'ynab' => [
+    'client_id' => env('YNAB_CLIENT_ID'),
+    'client_secret' => env('YNAB_CLIENT_SECRET'),
+    'redirect' => env('YNAB_CALLBACK_URL'),
+    'scope' => explode(",", env('YNAB_SCOPES'))
```
Comment (atymic) [inline src/YNAB/README.md:18]: Why is this needed?

File: src/YNAB/README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+```php
+return Socialite::driver('ynab')->redirect();
+```
+
+Example env
+```php
+YNAB_CLIENT_ID=abc123
+YNAB_CLIENT_SECRET=abc123
+YNAB_CALLBACK_URL=https://your-app.ngrok.io/oauth2/callback
+YNAB_LOGIN_SCOPE="read-only"
```
Comment (atymic) [inline src/YNAB/README.md:66]: This doesn't seem to be used

File: src/YNAB/YNABExtendSocialite.php [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+use SocialiteProviders\Manager\SocialiteWasCalled;
+
+class YNABExtendSocialite
+{
+    public function handle(SocialiteWasCalled $socialiteWasCalled): void
+    {
+        $socialiteWasCalled->extendSocialite('ynab', Provider::class);
+    }
+}
```
Comment (atymic) [inline src/YNAB/YNABExtendSocialite.php:13]: missing newline EOF

File: src/YNAB/composer.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            "email": "cameron@wilby.com"
+        }
+    ],
+    "support": {
+        "issues": "https://github.com/socialiteproviders/providers/issues",
+        "source": "https://github.com/socialiteproviders/providers",
+        "docs": "https://socialiteproviders.com/ynab"
+    },
+    "require": {
+        "php": "^8.0",
```
Comment (atymic) [inline src/YNAB/composer.json:24]:
```suggestion
        "php": "^8.1",
```

**Review by atymic — CHANGES_REQUESTED** [review]

File: src/YNAB/README.md (hunk 2 above)
Comment (cwilby) [reply src/YNAB/README.md:18]: It's not, will remove

File: src/YNAB/composer.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        "php": "^8.1",
+        "ext-json": "*",
+        "socialiteproviders/manager": "^4.4"
+    },
+    "autoload": {
+        "psr-4": {
+            "SocialiteProviders\\YNAB\\": ""
+        }
+    }
+}
```
Comment (atymic) [inline src/YNAB/composer.json:33]: sorry one more missing newline

Comment (cwilby) [reply src/YNAB/composer.json]: no worries thanks
